daily_milestone.go

```diff
@@ -1,29 +1,225 @@
+/*
+Copyright 2017 - The Dailymile Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
 package main
 
-import "flag"
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Struct to be used for project
+// project ....
+// Struct to be used for milestone
+// milestoneApi .....
+type milestoneAPI struct {
+	ID          int        `json:"id"`
+	Iid         int        `json:"iid"`
+	ProjectID   int        `json:"project_id"`
+	Title       string     `json:"title"`
+	Description string     `json:"description"`
+	State       string     `json:"state"`
+	CreatedAt   *time.Time `json:"created_at"`
+	UpdatedAt   *time.Time `json:"updated_at"`
+	StartDate   string     `json:"start_date"`
+	DueDate     string     `json:"due_date"`
+}
+type gitLabAPI struct {
+	ID          int        `json:"id"`
+	Iid         int        `json:"iid"`
+	ProjectID   int        `json:"project_id"`
+	Title       string     `json:"title"`
+	Description string     `json:"description"`
+	StartDate   string     `json:"start_date"`
+	DueDate     string     `json:"due_date"`
+	State       string     `json:"state"`
+	UpdatedAt   *time.Time `json:"updated_at"`
+	CreatedAt   *time.Time `json:"created_at"`
+	Name        string     `json:"name"`
+	NameSpace   struct {
+		ID       int    `json:"id"`
+		Name     string `json:"name"`
+		Path     string `json:"path"`
+		Kind     string `json:"kind"`
+		FullPath string `json:"full_path"`
+	} `json:"namespace"`
+}
+
+// Initialization of logging variable
+var logger *log.Logger
+
+// LoggerSetup Initialization
+func LoggerSetup(info io.Writer) {
+	logger = log.New(info, "INFO: ", log.Ldate|log.Ltime|log.Lshortfile)
+}
+
+// Function to get project ID from the gitLabAPI
+func getProjectID(baseURL string, Token string, Projectname string, Namespace string) (string, error) {
+	project := []gitLabAPI{}
+	urls := "https://" + baseURL + "/api/v4" + "/projects"
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", urls, nil)
+	if err != nil {
+		logger.Println(err)
+
+	}
+	req.Header.Add("PRIVATE-TOKEN", Token)
+	resp, err := client.Do(req)
+	if err != nil {
+		logger.Println(err)
+	}
+
+	respByte, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		logger.Println("fail to read response data")
+
+	}
+	json.Unmarshal(respByte, &project)
+	defer resp.Body.Close()
+	for _, p := range project {
```

Comment:
It's a list of projects so naming them project is leading in the code. Would call the list projectList or at least projects.